core/src/main/java/org/lflang/federated/generator/FedUtils.java

```diff
@@ -14,7 +14,17 @@ public static SupportedSerializers getSerializer(
     // Get the serializer
     SupportedSerializers serializer = SupportedSerializers.NATIVE;
     if (connection.getSerializer() != null) {
-      serializer = SupportedSerializers.valueOf(connection.getSerializer().getType().toUpperCase());
+      boolean isCustomSerializer = true;
+      for (SupportedSerializers method : SupportedSerializers.values()) {
+        if (method.name().equalsIgnoreCase(connection.getSerializer().getType())) {
+          serializer =
+              SupportedSerializers.valueOf(connection.getSerializer().getType().toUpperCase());
+          isCustomSerializer = false;
+        }
+        if (isCustomSerializer) {
+          serializer = SupportedSerializers.fromCustomString(connection.getSerializer().getType());
+        }
+      }
```

Comment:
The logic here doesn't look quite right.  If `connection.getSerializer()` returns something that doesn't match the first `method`, then you assume it is custom.  But shouldn't it not match any `method`?  Also, what if there is no serializer specified?  Does `connection.getSerializer()` return null? If so, this will give a null-pointer exception.

Reply:
Ah I made a mistake... The `if (isCustomSerializer)` should be outside the `for` loop. If there is no serializer specified, it will default to `NATIVE` because there is a null check before line 17 in the original code.
```java
SupportedSerializers serializer = SupportedSerializers.NATIVE;
    if (connection.getSerializer() != null) {
      boolean isCustomSerializer = true;
      for (SupportedSerializers method : SupportedSerializers.values()) {
        if (method.name().equalsIgnoreCase(connection.getSerializer().getType())) {
          serializer =
              SupportedSerializers.valueOf(connection.getSerializer().getType().toUpperCase());
          isCustomSerializer = false;
        }
      }
      if (isCustomSerializer) {
        serializer = SupportedSerializers.fromCustomString(connection.getSerializer().getType());
      }
    }
```

Reply:
OK, this looks better.  I suggest adding a `break` statement after `isCustomSerializer = false`.

Reply:
Added!